server/pulp/server/logs.py

```diff
@@ -84,6 +88,29 @@ def stop_logging():
     root_logger.handlers = []
 
 
+class TaskIDFilter(logging.Filter):
+    """
+    This is a filter which injects the current task id from celery into the log.
+    """
+    def filter(self, record):
+        record.task_id = get_current_task_id()
+        return True
+
+
+class TaskLogFormatter(logging.Formatter):
+    """
+    This formatter changes format depending on whether there is a task_id in
+    the LogRecord
+    """
+    def format(self, record):
+        if record.task_id is None:
+            self._fmt = LOG_FORMAT_STRING
+        else:
+            record.task_id = record.task_id[:8]
+            self._fmt = TASK_LOG_FORMAT_STRING
+        return super(TaskLogFormatter, self).format(record)
```

Comment:
Looks like in python 2.6 (RHEL 6) the `logging.Formatter` is not a new-style object so you cannot use `super()`.